core/corehttp/gateway_handler.go

```diff
@@ -50,7 +50,8 @@ func (i *gatewayHandler) newDagFromReader(r io.Reader) (*dag.Node, error) {
 	// return ufs.AddFromReader(i.node, r.Body)
 	return importer.BuildDagFromReader(
 		i.node.DAG,
-		chunk.DefaultSplitter(r))
+		chunk.DefaultSplitter(r),
+		nil)
```

Comment:
if we make the splitter change i mentioned earlier, we can move the 'add opts' into the constructor for the splitter, instead of this call.